### ISSUE TITLE
[question] DateTimePicker component saves browser date and time

### ISSUE THREAD

**Issue by Arkamit** [mui/mui-x#8461]
### Order ID or Support key 💳 (optional)

59954

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

We are using "@mui/x-date-pickers": "^5.0.17" and found that by default the DateTimePicker is picking up the local time(browser time). Is there a way we can configure the component for a specific timezone?

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
    Browser:
     Chrome: Version 111.0.5563.147 (Official Build) (64-bit)
    System:
    OS: Linux 4.14 Amazon Linux 2
    CPU: (4) x64 AMD EPYC 7571
    Memory: 12.47 GB / 15.52 GB
    Container: Yes
    Shell: 4.2.46 - /bin/bash
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    npm: 8.5.0 - /usr/local/bin/npm
  Managers:
    pip2: 18.0 - /usr/bin/pip2
    pip3: 20.2.2 - /usr/bin/pip3
    Yum: 3.4.3 - /usr/bin/yum
  Utilities:
    Make: 3.82 - /usr/bin/make
    Git: 2.32.0 - /usr/bin/git
  Virtualization:
    Docker: 20.10.13 - /usr/bin/docker
  IDEs:
    Nano: 2.9.8 - /usr/bin/nano
    VSCode: 1.77.0 - /home/adutta@huronconsultinggroup.com/.vscode-server/bin/7f329fe6c66b0f86ae1574c2911b681ad5a45d63/bin/remote-cli/code
    Vim: 8.2 - /usr/bin/vim
  Languages:
    Bash: 4.2.46 - /usr/bin/bash
    Perl: 5.16.3 - /usr/bin/perl
    Python: 2.7.18 - /usr/bin/python
    Python3: 3.7.10 - /usr/bin/python3
  Databases:
    SQLite: 3.7.17 - /usr/bin/sqlite3
```
</details>

**Comment by flaviendelangle** [comment]
Hi,

I'm currently working on a proper support for UTC / timzone in #8261
Which date library are you using for your pickers ?

**Comment by Arkamit** [comment]
Hi @flaviendelangle,

We are currently using Date-fns but we tried other libraries like Luxon, Dayjs and Moment and none of them worked fine.

Great to know that you are already looking into this issue.

**Comment by flaviendelangle** [comment]
The support for date-fns might be complicated, for now my PR focused on dayjs / moment and Luxon.

**Comment by Arkamit** [comment]
@flaviendelangle I wanted to mention that this is a bit critical for our project and our project uses @mui/x-date-pickers": "^5.0.17. So would it be possible to release a hot-fix for this?

**Comment by flaviendelangle** [comment]
The timezone / UTC support is not a hotfix, but a major evolution that will not be back-portable to v5.

**Comment by Arkamit** [comment]
@flaviendelangle Thanks for that information. It would be great of you can mention the version number to which we need to update.

**Comment by flaviendelangle** [comment]
Depending on the pace of the development, it might be released in 6.2 or 6.3.
It's not available yet, I'm working on it.

**Comment by Arkamit** [comment]
Thanks for the information @flaviendelangle
